src/check/canonicalize/NodeStore.zig

```diff
@@ -1202,6 +1207,180 @@ pub fn addStatement(store: *NodeStore, statement: CIR.Statement) CIR.Statement.I
     return @enumFromInt(@intFromEnum(node_idx));
 }
 
+/// Replaces an existing statement node in the store.
+pub fn replaceStatement(store: *NodeStore, idx: CIR.Statement.Idx, statement: CIR.Statement) void {
```

Comment:
Is this duplicating the logic now across both `replaceStatement` and `addStatement`?